### PR TITLE
[MLIR][Presburger] Implement PresburgerSpace::mergeAndAlignSymbols

### PR DESCRIPTION
**Pull request by iambrj**

### CODE REVIEW

File: mlir/lib/Analysis/Presburger/PresburgerSpace.cpp [hunk 1]
```diff
@@ -293,6 +294,39 @@ void PresburgerSpace::setVarSymbolSeperation(unsigned newSymbolCount) {
   // `identifiers` remains same.
 }
 
+void PresburgerSpace::mergeAndAlignVarKind(VarKind kind,
```
Comment (Groverkss) [inline mlir/lib/Analysis/Presburger/PresburgerSpace.cpp:297]: Do we need non symbol merging anywhere?

Comment (iambrj) [reply mlir/lib/Analysis/Presburger/PresburgerSpace.cpp:297]: I haven't seen any non-symbol merging in the affine dialect glue code.

Comment (Groverkss) [reply mlir/lib/Analysis/Presburger/PresburgerSpace.cpp:297]: Let's do mergeSymbols only then. I don't think we should have merging for any other variable kind.

**Comment by tobiasgrosser** [comment]
You probably want to adjust the title of the PR as well.

**Review by Groverkss — APPROVED** [review]
LGTM

**Pull request merged**